### PR TITLE
Adding docs config and requirements

### PR DESCRIPTION
**Pull request by hectorhdzg**
Local docs deployed available here:
https://hectorhdzg-opentelemetry-azure-python.readthedocs.io/en/latest/index.html

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -57,6 +57,7 @@ commands =
 [testenv:docs]
 deps =
   -c dev-requirements.txt
```
Comment (lzchen) [inline tox.ini]: Do we still need `dev-requirements.txt` in the `doc` env?

File: docs-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+sphinx~=2.4
+sphinx-rtd-theme~=0.4
+sphinx-autodoc-typehints~=1.10.2
+# External
+opentelemetry-api
+opentelemetry-sdk
+psutil
```
Comment (lzchen) [inline docs-requirements.txt]: I get errors if I don't include `psutil` but I can't seem to find where in the docs it is needed.

File: docs-requirements.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+sphinx~=2.4
+sphinx-rtd-theme~=0.4
+sphinx-autodoc-typehints~=1.10.2
+# External
+opentelemetry-api
```
Comment (lzchen) [inline docs-requirements.txt]: I believe requirements files are used to simply specify the version numbers of the installations. We should keep the actual installations in `deps`.

File: tox.ini (hunk 1 above)
Comment (hectorhdzg) [reply tox.ini]: Yes, at least pytest is needed

File: docs-requirements.txt (hunk 2 above)
Comment (hectorhdzg) [reply docs-requirements.txt]: psutil is used in PerformanceMetrics class, docs need all external libraries used in the code to be able to build

File: tox.ini (hunk 1 above)
Comment (hectorhdzg) [reply tox.ini]: Never mind looks like there is some duplication of packages in requirement files, will clean up a little bit

**Review by lzchen — APPROVED** [review]
Nice!

**Comment by lzchen** [comment]
https://travis-ci.org/github/microsoft/opentelemetry-azure-monitor-python/builds/667835526

**Pull request merged**